pkgs/development/cuda-modules/db/schema.nix

```diff
@@ -0,0 +1,413 @@
+            };
+            deprecated = mkOption {
+              type = bool;
+              default = false;
+              description = "License definition no longer used in Nixpkgs";
+            };
+            spdxId = mkOption {
+              type = nullOr str;
+              default = null;
+              description = "Optional SPDX Identifier";
+            };
+            url = mkOption {
+              type = nullOr str;
+              default =
+                let
+                  ensureComposable = base: if lib.hasSuffix "/" base then base else "${base}/";
+                  base_url = ensureComposable cudb.base_url;
+                  distribution_path = ensureComposable cudb.license.distribution_path.${name};
+                  license_path = cudb.license.license_path.${name};
+                in
+                "${base_url}${distribution_path}${license_path}";
+              defaultText = "\${base_url}\${distribution_path}\${license_path}";
+              description = "Web service for accessing license text";
+            };
+          };
+        }
+      );
+    in
+    {
+      # :: Str -> ()
+      product = mkOption {
+        type = SetOfStr;
+        default = { };
+        example = {
+          "cuda" = 1;
+          "cudnn" = 1;
+          "tensorrt" = 1;
+        };
+        description = "Corresponds to `release_product` in NVIDIA manifests";
+      };
+
+      # :: ProductName -> PName -> ProductVersion -> Option<PackageVersion>
+      release =
+        mkColumnOption cudb.product
+          {
+            type = attrsOf (attrsOf (nullOr str));
+            default = { };
+            description = ":: PName -> ProductVersion -> Option<PackageVersion>";
+          }
+          {
+            default = { };
+            description = ''
+              :: ProductName -> PName -> ProductVersion -> Option<PackageVersion>
+
+              Packages (`pname`, `version`) published in each release (`productName`, `productVersion)`, or, in NVIDIA manifests, `(release_product, release_label)`)'';
+          };
+
+      # :: PName -> Version -> Set<SHA256>
+      archive.bucket =
+        mkColumnOption cudb.package.pname
+          {
+            type = attrsOf SetOfStr;
+            default = { };
+            description = ":: Version -> SHA256 -> ()";
+          }
+          {
+            description = ":: PName -> Version -> SHA256 -> ()";
+          };
+      archive.sha256 = mkOption {
+        description = ":: SHA256 -> { URL, SystemStringNvidia, CompatibilityTag -> () }";
+        type = attrsOf (submodule {
+          options = {
+            systemNv = mkOption {
+              type = enum (builtins.attrNames cudb.system.nvidia);
+              description = ":: SystemStringNvidia";
+            };
+            tags = mkOption {
+              type = SetOfStr;
+              default = { };
+              description = ":: CompatibilityTag -> ()";
```

Comment:
This type is terrifying, why does it return void? Or is `()` the same as `Unit`?

If it's the latter, I'd recommend writing

```
:: AttrSet CompatibilityTag Unit
```

to make more clear that it's not a function (I mean, you could add `__functor` but I don't think you want to do that).

Reply:
> Or is () the same as Unit?

Yeah like empty tuple?

Reply:
> to make more clear that it's not a function (I mean, you could add __functor but I don't think you want to do that).

I was considering writing `=>` to denote a hashmap (a "memoized function") but that's probably even more obscure